input/pages/index.md

```diff
@@ -0,0 +1,3304 @@
+
+> Note: Many of these functions will result in a set of nodes of different underlying types. It may be necessary to use `ofType()` as described in the previous section to maintain type safety. See [Type safety and strict evaluation](#type-safety-and-strict-evaluation) for more information about type safe use of FHIRPath expressions.
+
+### Utility functions
+
+#### trace(name : String [, projection: Expression]) : collection
+
+Adds a String representation of the input collection to the diagnostic log, using the `name` argument as the name in the log. This log should be made available to the user in some appropriate fashion. Does not change the input, so returns the input collection as output.
+
+If the `projection` argument is used, the trace would log the result of evaluating the project expression on the input, but still return the input to the trace function unchanged.
+
+``` fhirpath
+contained.where(criteria).trace('unmatched', id).empty()
+```
+
+The above example traces only the id elements of the result of the where.
+
+#### Current date and time functions
+
+The following functions return the current date and time. The timestamp that these functions use is an implementation decision, and implementations should consider providing options appropriate for their environment. In the simplest case, the local server time is used as the timestamp for these function.
+
+To ensure deterministic evaluation, these operators should return the same value regardless of how many times they are evaluated within any given expression (i.e. now() should always return the same DateTime in a given expression, timeOfDay() should always return the same Time in a given expression, and today() should always return the same Date in a given expression.)
+
+##### now() : DateTime
+
+Returns the current date and time, including timezone offset.
+
+##### timeOfDay() : Time
+
+Returns the current time.
+
+##### today() : Date
+
+Returns the current date.
+
+<a name="definevariable"></a>
+#### defineVariable(name: String [, expr: expression])
+> Note: The contents of this section are Standard for Trial Use (STU)
+{: .stu-note }
+
+Defines a variable named `name` that is accessible in subsequent expressions and has the value of `expr` if present, otherwise the value of the input collection. In either case the function does not change the input and the output is the same as the input collection.
+{:.stu}
+
+If the name already exists in the current expression scope, the evaluation will end and signal an error to the calling environment. Note that functions that take an `expression` as an argument establish a scope for the iteration variables ($this and $index). If a variable is defined within such an expression, it is only available within that expression scope.
```

Comment:
"Note that functions that take an expression as an argument establish a scope...".  That would suggest that defineVariable also establishes a scope, since it takes an expression argument, but I don't think it does.

Reply:
I believe that you're correct here, there's another tracker that's specifically walking into the whole under-spec'd scope part - including the addition of a new $focus variable.